retrofit/src/main/java/retrofit/MethodInfo.java

```diff
@@ -85,8 +86,8 @@
     this.method = method;
     this.adapterFactory = adapterFactory;
     this.converterFactory = converterFactory;
-    parseResponseType();
     parseMethodAnnotations();
+    parseResponseType();
```

Comment:
This gross parsing order dependency on this highly-mutable object will die soon.